chanzuckerberg/edu-design-system
# feat: add scss, js, json transforms to style dictionary

### Summary:

[ch187479]

Add back transforms from `main` (previously skipped b/c they caused build errors).

Turns out the error was from outputting built `json` files into `design-tokens/`, because then style-dictionary was trying to read those files as tokens again.

Instead, this:
- creates a new `tokens-dist/` folder to explicitly store built tokens
- copies `tokens-dist/` into `design-tokens/` during the build

Note that this flattens the imports 1 level -- I feel like it's been annoying to have to import from e.g. `design-tokens/json/variables.json` instead of `design-tokens/variables.json`. It's a change from our current approach tho so open to change on the output path

### Test Plan:
- run `yarn build:tokens`
- run `yarn start` to load storybook
- run `yarn build`, confirm built tokens are there:
![Screen Shot 2022-03-31 at 3 32 28 PM](https://user-images.githubusercontent.com/15840841/161134768-a6703d45-22e2-4d89-8484-aa2d6c98c9a8.png)
- Note that `design-tokens/` is no longer in the built package, which I think is correct
